x-pack/legacy/plugins/siem/public/components/markdown_editor/index.tsx

```diff
@@ -41,81 +57,116 @@ const Container = styled(EuiPanel)`
   `}
 `;
 
-const Tabs = styled(EuiTabbedContent)`
-  width: 100%;
-`;
-
-const Footer = styled(EuiFlexGroup)`
-  ${({ theme }) => css`
-    height: 41px;
-    padding: 0 ${theme.eui.euiSizeM};
-    .euiLink {
-      font-size: ${theme.eui.euiSizeM};
-    }
-  `}
-`;
-
 const MarkdownContainer = styled(EuiPanel)`
   min-height: 150px;
   overflow: auto;
 `;
 
+export interface CursorPosition {
+  start: number;
+  end: number;
+}
+
 /** An input for entering a new case description  */
 export const MarkdownEditor = React.memo<{
-  placeholder?: string;
-  footerContentRight?: React.ReactNode;
+  bottomRightContent?: React.ReactNode;
+  topRightContent?: React.ReactNode;
   initialContent: string;
   isDisabled?: boolean;
   onChange: (description: string) => void;
-}>(({ placeholder, footerContentRight, initialContent, isDisabled = false, onChange }) => {
-  const [content, setContent] = useState(initialContent);
-  useEffect(() => {
-    onChange(content);
-  }, [content]);
-  const tabs = useMemo(
-    () => [
-      {
-        id: 'comment',
-        name: i18n.MARKDOWN,
-        content: (
-          <TextArea
-            onChange={e => {
-              setContent(e.target.value);
-            }}
-            aria-label={`markdown-editor-comment`}
-            fullWidth={true}
-            disabled={isDisabled}
-            placeholder={placeholder ?? ''}
-            spellCheck={false}
-            value={content}
-          />
-        ),
-      },
-      {
-        id: 'preview',
-        name: i18n.PREVIEW,
-        content: (
-          <MarkdownContainer data-test-subj="markdown-container" paddingSize="s">
-            <Markdown raw={content} />
-          </MarkdownContainer>
-        ),
-      },
-    ],
-    [content, isDisabled, placeholder]
-  );
-  return (
-    <Container>
-      <Tabs data-test-subj={`markdown-tabs`} size="s" tabs={tabs} initialSelectedTab={tabs[0]} />
-      <Footer alignItems="center" gutterSize="none" justifyContent="spaceBetween">
-        <EuiFlexItem grow={false}>
-          <EuiLink href={MARKDOWN_HELP_LINK} external target="_blank">
-            {i18n.MARKDOWN_SYNTAX_HELP}
-          </EuiLink>
-        </EuiFlexItem>
-        {footerContentRight && <EuiFlexItem grow={false}>{footerContentRight}</EuiFlexItem>}
-      </Footer>
-    </Container>
-  );
-});
+  onCursorPositionUpdate?: (cursorPosition: CursorPosition) => void;
+  placeholder?: string;
+}>(
+  ({
+    bottomRightContent,
+    topRightContent,
+    initialContent,
+    isDisabled = false,
+    onChange,
+    placeholder,
+    onCursorPositionUpdate,
+  }) => {
+    const [content, setContent] = useState(initialContent);
+    useEffect(() => {
+      onChange(content);
+    }, [content]);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const setCursorPosition = useCallback((e: any) => {
```

Comment:
i cannot figure out how to get `EventTarget` and `HTMLTextAreaElement` combined!

Reply:
that seems to do the trick
```
const setCursorPosition = useCallback((e: ChangeEvent<HTMLInputElement>) => {
      if (onCursorPositionUpdate) {
        onCursorPositionUpdate({
          start: e.target.selectionStart ?? 0,
          end: e.target.selectionEnd ?? 0,
        });
      }
    }, []);
```

Reply:
I do not know If I am missing something here but I tried to just `onBlur` props in the component `TextArea` with this handler and I did not see any problems

```
const setCursorPosition = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
      if (onCursorPositionUpdate) {
        onCursorPositionUpdate({
          start: e!.target!.selectionStart ?? 0,
          end: e!.target!.selectionEnd ?? 0,
        });
      }
      return false;
    };
 .........
<TextArea
    onChange={e => {
    setContent(e.target.value);
    }}
    onBlur={setCursorPosition}
     aria-label={`markdown-editor-comment`}
     fullWidth={true}
     disabled={isDisabled}
     placeholder={placeholder ?? ''}
     spellCheck={false}
     value={content}
/>
```

Reply:
im an idiot, was working off of Eui docs and didnt see `onBlur` in props so found that silly way to blur